components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java

```diff
@@ -188,4 +379,125 @@ public void testBuildURLWithQueryParams(String url, Map<String, String> queryPar
+        cookies[1] = new Cookie(FrameworkConstants.COMMONAUTH_COOKIE, "commonAuthIdValue");
+        cookies[1].setPath(FrameworkConstants.TENANT_CONTEXT_PREFIX + TENANT_DOMAIN + "/");
+        return cookies;
+    }
+
+    private void mockCookieTest() {
+        mockStatic(SameSiteCookie.class);
+        Cookie[] cookies = getAuthenticationCookies();
+        when(request.getCookies()).thenReturn(cookies);
+    }
+
+    @Test
+    public void testRemoveAuthCookie() {
+        mockCookieTest();
+
+        FrameworkUtils.removeAuthCookie(request, response);
+        verify(response, times(1)).addCookie(cookieCaptor.capture());
+        List<Cookie> capturedCookies = cookieCaptor.getAllValues();
+        Cookie removedOldSessionNonce = capturedCookies.get(0);
+
+        assertEquals(removedOldSessionNonce.getName(), FrameworkConstants.COMMONAUTH_COOKIE);
+    }
+
+    @Test
+    public void testRemoveAuthCookieInTenant() {
+        mockCookieTest();
+        FrameworkUtils.removeAuthCookie(request, response, TENANT_DOMAIN);
+        verify(response, times(1)).addCookie(cookieCaptor.capture());
+        List<Cookie> capturedCookies = cookieCaptor.getAllValues();
+
+        Cookie removedCookie = capturedCookies.get(0);
+        assertEquals(removedCookie.getName(), FrameworkConstants.COMMONAUTH_COOKIE);
+        assertEquals(removedCookie.getPath(), FrameworkConstants.TENANT_CONTEXT_PREFIX + TENANT_DOMAIN + "/");
+    }
+
+    @Test
+    public void testRemoveCookieNonExistCookieConfig() {
+        mockCookieTest();
+        FrameworkUtils.removeCookie(request, response, FrameworkConstants.COMMONAUTH_COOKIE, SameSiteCookie.STRICT, ROOT_DOMAIN);
+
+        verify(response, times(1)).addCookie(cookieCaptor.capture());
+        List<Cookie> capturedCookies = cookieCaptor.getAllValues();
+        Cookie removedCookie = capturedCookies.get(0);
+
+        assertEquals(removedCookie.getName(), FrameworkConstants.COMMONAUTH_COOKIE);
+        assertEquals(removedCookie.getPath(), ROOT_DOMAIN);
+        assertEquals(removedCookie.getMaxAge(), 0);
+    }
+
+    @Test
+    public void testRemoveCookieExistCookieConfig() {
+        mockCookieTest();
+
+        IdentityCookieConfig cookieConfig = new IdentityCookieConfig(FrameworkConstants.COMMONAUTH_COOKIE);
+        mockStatic(IdentityUtil.class);
+        when(IdentityUtil.getIdentityCookieConfig(FrameworkConstants.COMMONAUTH_COOKIE)).thenReturn(cookieConfig);
+
+        FrameworkUtils.removeCookie(request, response, FrameworkConstants.COMMONAUTH_COOKIE,
+                SameSiteCookie.STRICT, ROOT_DOMAIN);
+
+        verify(response, times(1)).addCookie(cookieCaptor.capture());
+        List<Cookie> capturedCookies = cookieCaptor.getAllValues();
+        Cookie removedCookie = capturedCookies.get(0);
+
+        assertEquals(removedCookie.getName(), FrameworkConstants.COMMONAUTH_COOKIE);
+        assertEquals(removedCookie.getPath(), ROOT_DOMAIN);
+        assertEquals(removedCookie.getMaxAge(), 0);
+
```

Comment:
Let's remove this new line.

Reply:
Addressed with commit 1e2bf847a014fe13c6c7af5d03b6cc4280e6bdec in https://github.com/wso2/carbon-identity-framework/pull/3774 PR